src/main/js/helpers/legend.js

```diff
@@ -92,27 +94,100 @@ const loadLegendItem = (legendSVG, t, shownTargets, eventHandlers) => {
                 eventHandlers.hoverHandler(t, constants.HOVER_EVENT.MOUSE_EXIT)
             );
     }
-    itemPath
+    const buttonPath = itemPath
         .append("button")
         .classed(styles.legendItemBtn, true)
         .attr("title", text)
         .attr("tabindex", shouldForceDisableLegendItem ? -1 : 0)
-        .append(() =>
-            new Shape(getShapeForTarget(t)).getShapeElement(
-                getDefaultSVGProps({
-                    svgClassNames: styles.legendItemIcon,
-                    svgStyles: `fill: ${getColorForTarget(t)};`
-                }),
-                true
-            )
-        );
+        .append("span")
+        .attr("class", styles.legendItemSpan);
+
+    processLegendOptions(buttonPath, t);
+
     itemPath
         .append("label")
         .classed(styles.legendItemText, true)
         .attr("tabindex", -1)
         .text(text);
     return legendSVG;
 };
+
+/**
+ * Creates legend button content based on legend options
+ *
+ * @private
+ * @param {object} buttonPath - d3 svg object
+ * @param {object} t - input item object processed from the input JSON
+ */
+const processLegendOptions = (buttonPath, t) => {
+    if (t.legendOptions) {
+        if (t.legendOptions.showShape) {
+            createLegendIcon(buttonPath, t);
+        }
+        if (t.legendOptions.showLine) {
+            createLegendLine(buttonPath, t);
+        }
+    } else {
+        createLegendIcon(buttonPath, t);
+    }
+};
+
+/**
+ * Creates an icon in the legend button
+ *
+ * @private
+ * @param {object} buttonPath - d3 svg object
+ * @param {object} t - input item object processed from the input JSON
+ */
+const createLegendIcon = (buttonPath, t) => {
+    buttonPath.append(() =>
```

Comment:
Remove the braces since you are saying that you are returning an object in JSDoc

Reply:
Updated